plugins/woocommerce-admin/client/tasks/task-list-item.tsx

```diff
@@ -133,16 +135,30 @@ export const TaskListItem: React.FC< TaskListItemProps > = ( {
 		if ( actionUrl ) {
 			if ( actionUrl.startsWith( 'http' ) ) {
 				window.location.href = actionUrl;
-			} else {
-				( getHistory() as History ).push(
+				return;
+			}
+
+			if ( ! isWCAdmin( window.location.href ) ) {
```

Comment:
It did occur to me that it may make sense to have a utility that navigates to a page, utilizing the router if appropriate, or redirecting if not (as we're doing here). Is that more-or-less what you had in mind @joshuatf ?

Reply:
Exactly!  This seems to be a pretty common occurrence and it would be great to have both a utility and test around it for use in all our components.  The navigation package seems like a good candidate for its home.

Reply:
Consider it done! Elaborated in my [other comment](https://github.com/woocommerce/woocommerce/pull/33124#issuecomment-1137877412), so let me know what you think.